wallet/src/de/schildbach/wallet/data/BlockchainStateDao.kt

```diff
@@ -30,7 +31,7 @@ import androidx.room.Query
 abstract class BlockchainStateDao {
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    abstract fun insert(blockchainState: BlockchainState)
+    protected abstract fun insert(blockchainState: BlockchainState)
```

Comment:
Given that `save` has additional checks, it's better to hide this method so that no one uses it by mistake.